ToDoList/src/ToDoList.WebApi/Program.cs

```
@@ -8,10 +9,18 @@
     builder.Services.AddControllers();
     builder.Services.AddSwaggerGen();
     builder.Services.AddDbContext<ToDoItemsContext>();
-    builder.Services.AddScoped<IRepository<ToDoItem>, ToDoItemsRepository>(); // kdykoliv se odkazujina IRepository<ToDoItem>, tak pouzij ToDoItemRepository ~~ Dependency Injection
+
+    builder.Services.AddScoped<IRepositoryAsync<ToDoItem>, ToDoItemsRepository>(); // Dependency Injection
 }
 
 var app = builder.Build();
+
+using (var scope = app.Services.CreateScope()) // neslo mi provest migraci, s pomoci chatGPT jsem se dobrala k tomu, že problém byl v migraci v konstruktoru ToDoItemsContextu
```

migrace mi padala na

Unable to create a 'DbContext' of type 'RuntimeType'. The exception 'An error was generated for warning 'Microsoft.EntityFrameworkCore.Migrations.PendingModelChangesWarning': The model for context 'ToDoItemsContext' has pending changes. Add a new migration before updating the database. See https://aka.ms/efcore-docs-pending-changes. This exception can be suppressed or logged by passing event ID 'RelationalEventId.PendingModelChangesWarning' to the 'ConfigureWarnings' method in 'DbContext.OnConfiguring' or 'AddDbContext'.' was thrown while attempting to create an instance. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728

přesun db.Database.Migrate() sem z dbContextu to vyřešil. Ale budu ráda za tvůj názor a nějaký komentář, jestli to jde řešit jinak, případně jak.

---

Myslím, že do constructoru to dal Lektor kvůli úsporu času při lekci. Protože ten constructor se může zavolat i několikrát během volání aplikace.

Jako Java vývojář doporučuji tohle všechno dávat do startu programu. Aby se to vše provedlo před startem aplikace a pak vše fungovalo jak má. Tady v C# to vidím obdobně. Také to dát do Program.cs, kde při startu programu se řeší dependence, frontend, tak i init či migrace databáze. A constructor bych pak už nechal prázdný. 

Dobrá práce. Jdeš správným směrem :)